### ISSUE TITLE
bug: error when running yarn - cmake: command not found

### ISSUE THREAD

**Issue by TillaTheHun0** [hyper63/hyper63#198]
### Stacktrace:
```
/Users/tyler/Development/hyper/hyper63/node_modules/zeromq/build/Release/../../script/build.sh: line 60: cmake: command not found
make: *** [libzmq/lib] Error 127
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/tyler/.nvm/versions/node/v16.2.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:365:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
gyp ERR! System Darwin 20.3.0
gyp ERR! command "/Users/tyler/.nvm/versions/node/v16.2.0/bin/node" "/Users/tyler/.nvm/versions/node/v16.2.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/tyler/Development/hyper/hyper63/node_modules/zeromq
```

### Steps to reproduce:

run `yarn` to install dependencies

### Background:

When installing dependencies using `yarn`, I encountered the error above. `zeromq`, a dependency of the `adapter-zeromq` package, has a postinstall step that requires `cmake` to be available. However, OSX does not have `cmake` installed by default and thus causes an error.

Gitpod installs `cmake` as part of the [image](https://github.com/gitpod-io/workspace-images/blob/b1c94e57ab0dfc06594abc2872815dc7de91e6ba/full/Dockerfile#L73) it uses to create development environments, so we don't run into this issue there.

### Solution

I doubt core team will run into this issue, since we primarily use Gitpod, but contributors who develop locally may run into this issue. We should add installing `cmake` to contributor documentation or figure out another solution.

### Workaround

Installing `cmake` resolved the issue.

**Comment by twilson63** [comment]
Maybe we should add a note that mac users should install xcode command developer tools, which should include all of the common sdks for development needs.

https://developer.apple.com/downloads/

XCode Command Line Tools

Or consider dropping adapters from the main repo and placing them in separate repos since they will not be required for all implementations. Maybe we should separate the adapters into separate repos with their specific dependencies, because as we implement more adapters, we are likely to encounter more specific dependencies like zeromq. That everyone that may be contributing does not want to deal with.

What are your thoughts, @tripott @TillaTheHun0 ?

**Comment by TillaTheHun0** [comment]
> Maybe we should separate the adapters into separate repos with their specific dependencies, because as we implement more adapters, we are likely to encounter more specific dependencies like zeromq. That everyone that may be contributing does not want to deal with.

That thought came to mind for me as well. It seems like [other folks](https://stackoverflow.com/questions/60874627/how-to-avoid-install-of-packages-in-monorepo-using-yarn) have run into this as well.

I _really_ lean towards monorepos, but perhaps some adapters need to be in separate repos. I'll investigate if there's a way to get around this in "monorepo lingua franca"

**Comment by TillaTheHun0** [comment]
closed by #232
